[Test Case]: Alignment Infrastructure Curves (#26)
bSI-InfraRoom/MVD-Infra-Test-Instructions: E1a-ARSE/ALIN/README.md

# Test Instruction

| Documentation Code  | Title                           | Exchange Code | Test Code | Author        | Data Owner | Version | Date       |
| ------------------- | ------------------------------- | ------------- | --------- | ------------- | ---------- | ------- | ---------- |
| IFC4.3AbRV_E1a_ALIN | Alignment Infrastructure Curves | E1a           | ALIN      | Lars Wikström | FTIA?      | 1.0     | 07.01.2022 |

## Summary (Intent)

*Include a short description of the test case. This description should include a summary of the capabilities and data representations being tested by the defined data set.*

> With these instructions the infrastructure (e.g. Road) alignment exchange will be established.This includes basic alignment geometry featuring the common layout:

Reviewer: ```suggestion
With these instructions the infrastructure (e.g. Road) alignment exchange is established.
This includes basic alignment geometry featuring the common layout:
```

- horizontal
- vertical

The valid geometry segment types for horizontal are the following:

- Arc
- Clothoid
- Linear
- *Check if other transition curves are being used*

The valid geometry segment types for vertical:

- Linear

> - Arc 

Reviewer: ```suggestion
- Circular Arc 
```

I'd be more explicit.

- *Alignment Layout*
- *Alignment Geometry*
- *Alignment Geometry Gradient*
- *Project Global Positioning*
- *Spatial Containment*

</details>

## Test Case Imports
Test instructions are defined with a modular approach to reduce repetition of validation criteria and test content, and improve vendors ability to solve issues and bugs. therefore this test instruction *imports/reuses* the following Test instructions and entities with the relevant associated validation criteria.

:construction: under construction :construction:

<details><summary>Imports & Reuses</summary>

| TI Code                                  | Test Instruction Title  | Comments                                                     |
| ---------------------------------------- | ----------------------- | ------------------------------------------------------------ |
| [IFC4.3AbRV_E0_SSRD](../../E0-SSRD/SSRD) | Spatial Structures Road | Spatial structure for road incuding the dependencies (E0_SSSI, E0_MSTP) |

> 

Reviewer: Do we have a georeferencing TI?

Author: Thanks for commenting @pjanck! I have assumed that georeferencing is part of E0_MSTP which is mentioned in the table (and if you take a look into the TI dependency diagram from @AlexBrad1eyCT) on Miro https://miro.com/app/board/o9J_lkyDq70=/.
As you say, there's a lot still to be defined. My main issues are:
- I have assumed straight line, circular arc, clothoid for horizontal and straight line, parabolic or circular arc for vertical. Is that all for a road alignment or are there other more obscure cases?
- Is it enough to test straight, left turn, right turn in horizontal and straight up/down (connected with arcs) for vertical?
- How to produce the actual test case? I looked at e.g. the alignment from MCON-2 but that does not contain clothoids. It would be nice to have a set case that uses all necessary segment types in all typical ways (left/right, up/down, transitions, ...).

Reviewer 2: For a road alignment, even parabolic arc is rare enough to raise the question: is it going to be mandatory for certification? Looking at validation criteria "For this test case to be considered passed **all capabilities** listed in this section shall be verified, with no exception" and "At least 1 instance of each entity listed in Itemised Roots is present in the file", I'd be very careful with what is included in the case (should be testing IFC export/import, not software capability in general). If it's not against any chosen certification strategy (?) might be reasonable to allow also "passed with exceptions (documented)", but if the validation criteria for a case really must be that draconian, I'd suggest making first a basic road alignment case with linear, circular arc and clothoid for horizontal; linear and circular arc for vertical.

Author: During my "peak days", as a developer of road design software (this was a long time ago), the only vertical "curved curve" used was parabolic arc. Unfortunately, I didn't find anything regarding this in the RA. Might have missed it though.  I wanted to mention both circular and parabolic until more info available

Reviewer 2: @larswik I'm not saying that parabolic arc wouldn't be valid vertical curve on roads and shouldn't be included in some test case. I'm just asking does it have to be supported as a mandatory requirement? If a software doesn't do parabolic arcs since their users do not use them they should still be able to get certified - I hope.

Author: @JuhaHy good point of course! But it goes both ways right? I'm not sure how to organize these variations, but let's see.

Reviewer: Meta: I'd move this discussion to a separate issue for easier traceability. Searching for these comments in the future will be a pain, as well as a separate issue enables others to chime in as well. @larswik I'll copy the comments if that is ok with you.

> georeferencing is part of E0_MSTP which is mentioned

That answers my question and resolves this conversation.

# Test Instruction

| Documentation Code  | Title                           | Exchange Code | Test Code | Author        | Data Owner | Version | Date       |
| ------------------- | ------------------------------- | ------------- | --------- | ------------- | ---------- | ------- | ---------- |
| IFC4.3AbRV_E1a_ALIN | Alignment Infrastructure Curves | E1a           | ALIN      | Lars Wikström | FTIA?      | 1.0     | 07.01.2022 |

## Summary (Intent)

*Include a short description of the test case. This description should include a summary of the capabilities and data representations being tested by the defined data set.*

With these instructions the infrastructure (e.g. Road) alignment exchange is established.
This includes basic alignment geometry featuring the common layout:

- horizontal
- vertical

The valid geometry segment types for horizontal are the following:

> - Arc

Reviewer 3: Would we just add parabolic arc here since the rest would be covered in E1?

Author: I think that we could do it like that.

Reviewer 3: ping @pjanck
